samples/samples/backup_sample_test.py

```diff
@@ -113,3 +119,13 @@ def test_cancel_backup(capsys):
         "Backup deleted." in out
     )
     assert cancel_success or cancel_failure
+
+
+@RetryErrors(exception=DeadlineExceeded, max_tries=2)
+def test_create_database_with_retention_period(capsys):
+    database_id = unique_database_id()
+    retention_period = "7d"
+    backup_sample.create_database_with_version_retention_period(INSTANCE_ID, database_id, retention_period)
+    out, _ = capsys.readouterr()
+    assert (database_id + " created with ") in out
+    assert ("retention period " + retention_period) in out
```

Comment:
Since there is no sample which deletes this database the cleanup needs to be done in the test. If you add `spanner_instance` to the function args: 
```
def test_create_database_with_retention_period(capsys, spanner_instance):
```
you can do this in two lines:
```
database = spanner_instance.database(database_id)
database.drop()
```